src/app/child-dev-project/notes/note-details/note-details.component.html

```diff
@@ -118,25 +118,53 @@ <h1>{{ entity.date?.toLocaleDateString() }}: {{ entity.subject }}</h1>
       </div>
     </div>
 
-    <div fxLayout="row wrap" fxLayoutGap="20px">
-      <div fxFlex *ngIf="!entity.category.isMeeting">
-        <app-child-select
-          [selectedChildrenIds]="entity.children"
-          (newIdAdded)="entity.addChild($event); entityForm.form.markAsDirty()"
-          (idRemoved)="
-            entity.removeChild($event); entityForm.form.markAsDirty()
-          "
-          class="input-medium"
+    <div fxLayoutGap="20px">
+
+      <app-entity-select
+        [entityType]="Child"
+        [entityView]="childBlock"
+        [(selection)]="entity.children"
+        (selectionChange)="entityForm.form.markAsDirty()"
+        [additionalFilter]="filterInactiveChildren"
+        label="Children"
+        placeholder="Add Child...">
+        <!-- Header for autocomplete -->
+        <mat-option (click)="toggleIncludeInactiveChildren()">
+          <mat-checkbox
+            labelPosition="before"
+            [checked]="includeInactiveChildren">
+            Include Inactive Children
+          </mat-checkbox>
+        </mat-option>
+      </app-entity-select>
+
+      <div *ngIf="entity.category.isMeeting">
+        <app-child-meeting-note-attendance
+          *ngFor="let childId of entity.children"
+          [childId]="childId"
+          [attendance]="entity.getAttendance(childId)"
+          (change)="entityForm.form.markAsDirty()"
+          (remove)="entity.removeChild(childId); entityForm.form.markAsDirty()"
         >
-        </app-child-select>
+        </app-child-meeting-note-attendance>
       </div>
-      <app-note-presence-list
-        *ngIf="entity.category.isMeeting"
-        [recordForm]="entityForm"
-        [entity]="entity"
-        fxFlex
-      >
-      </app-note-presence-list>
+
     </div>
+
   </form>
 </app-form-dialog-wrapper>
+
+<ng-template
```

Comment:
could we use content projection for these templates as well? (maybe with "multi-slot" approach, if necessary https://imm9o.medium.com/angular-content-projection-the-complete-guide-45d3bc1692f2)

Reply:
something I had considered. Unfortunately, I don't see/know how to pass the entities to the underlying module (currently this is done using the 'let-entity' approach in the template). I am happy to exchange this approach for a template since that's cleaner and more intuitive. However, I don't know exactly how to do that. Maybe there is a way to use something similar to what angular uses for table cells?